pkg/gardenlet/operation/botanist/monitoring.go

```diff
@@ -376,3 +457,67 @@ func (b *Botanist) getMonitoringComponents() []component.MonitoringComponent {
 
 	return monitoringComponents
 }
+
+// TODO(rfranzke): Remove this function after v1.100 has been released.
+func normalizeAlertingRules(input string) []string {
+	var cleanedLines []string
+	for _, line := range strings.Split(input, "\n") {
+		if strings.Contains(line, ": |") {
+			continue
+		}
+		if len(line) >= 2 && line[:2] == "  " {
+			cleanedLines = append(cleanedLines, line[2:])
+		} else {
+			cleanedLines = append(cleanedLines, line)
+		}
+	}
+
+	var blocks []string
+	for _, group := range strings.Split(strings.Join(cleanedLines, "\n"), "groups:") {
+		group = strings.TrimSpace(group)
+		if group != "" {
+			blocks = append(blocks, "groups:\n"+group)
+		}
+	}
+
+	return blocks
+}
```

Comment:
The string arithmetic done in this function is not directly obvious. Furthermore, it is not really clear why this is done. Normally, I would have voted for some documentation why this is required or some examples. However, as this is only temporary I can also live with it as is.
How do you see this?

Reply:
I agree, but in the light of temporary code, I hope we can live with it. This is really ugly, but I would like to refrain from investing more than needed for this.

Reply:
Fine with me.